zhaquirks/bosch/rbsh_trv0_zb_eu.py

```diff
@@ -0,0 +1,488 @@
+REMOTE_TEMPERATURE_ATTR_ID = 0x4040
+
+# Window open switch (changes to a lower target temperature when on).
+WINDOW_OPEN_ATTR_ID = 0x4042
+
+# Boost preset mode.
+BOOST_ATTR_ID = 0x4043
+
+"""Bosch specific user interface attribute ids."""
+
+# Display orientation with values BoschDisplayOrientation.
+SCREEN_ORIENTATION_ATTR_ID = 0x400B
+
+# Displayed temperature with values BoschDisplayedTemperature.
+DISPLAY_MODE_ATTR_ID = 0x4039
+
+# Display on-time (5s - 30s).
+SCREEN_TIMEOUT_ATTR_ID = 0x403A
+
+# Display brightness (0 - 10).
+SCREEN_BRIGHTNESS_ATTR_ID = 0x403B
+
+# Control sequence of operation (heating/cooling)
+CTRL_SEQUENCE_OF_OPERATION_ID = 0x001B
+
+
+class BoschOperatingMode(t.enum8):
+    """Bosh operating mode attribute values."""
+
+    Schedule = 0x00
+    Manual = 0x01
+    Pause = 0x05
+
+
+class State(t.enum8):
+    """Binary attribute (window open) value."""
+
+    Off = 0x00
+    On = 0x01
+
+
+class BoschDisplayOrientation(t.enum8):
+    """Bosch display orientation attribute values."""
+
+    Normal = 0x00
+    Flipped = 0x01
+
+
+class BoschDisplayedTemperature(t.enum8):
+    """Bosch displayed temperature attribute values."""
+
+    Target = 0x00
+    Measured = 0x01
+
+
+class BoschControlSequenceOfOperation(t.enum8):
+    """Supported ControlSequenceOfOperation modes."""
+
+    Cooling = ControlSequenceOfOperation.Cooling_Only
+    Heating = ControlSequenceOfOperation.Heating_Only
+
+
+"""HA thermostat attribute that needs special handling in the Bosch thermostat entity."""
+SYSTEM_MODE_ATTR = Thermostat.AttributeDefs.system_mode
+
+"""Bosch operating mode to HA system mode mapping."""
+OPERATING_MODE_TO_SYSTEM_MODE_MAP = {
+    BoschOperatingMode.Schedule: Thermostat.SystemMode.Auto,
+    BoschOperatingMode.Manual: Thermostat.SystemMode.Heat,
+    BoschOperatingMode.Pause: Thermostat.SystemMode.Off,
+}
+
+"""HA system mode to Bosch operating mode mapping."""
+SYSTEM_MODE_TO_OPERATING_MODE_MAP = {
+    Thermostat.SystemMode.Off: BoschOperatingMode.Pause,
+    Thermostat.SystemMode.Heat: BoschOperatingMode.Manual,
+    Thermostat.SystemMode.Cool: BoschOperatingMode.Manual,
+    Thermostat.SystemMode.Auto: BoschOperatingMode.Schedule,
+}
+
+"""Bosch display orientation enum to uint8_t mapping."""
+DISPLAY_ORIENTATION_ENUM_TO_INT_MAP = {
+    BoschDisplayOrientation.Normal: 0x00,
+    BoschDisplayOrientation.Flipped: 0x01,
+}
+
+
+class BoschThermostatCluster(CustomCluster, Thermostat):
+    """Bosch thermostat cluster."""
+
+    class AttributeDefs(Thermostat.AttributeDefs):
+        """Bosch thermostat manufacturer specific attributes."""
+
+        operating_mode = ZCLAttributeDef(
+            id=OPERATING_MODE_ATTR_ID,
+            type=BoschOperatingMode,
+            is_manufacturer_specific=True,
+            name="operating_mode",
+        )
+
+        pi_heating_demand = ZCLAttributeDef(
+            id=VALVE_POSITION_ATTR_ID,
+            # Values range from 0-100
+            type=t.enum8,
```

Comment:
We now support `zcl_type=foundation.DataTypeId.map8`, which would let you change it to `type=t.uint8_t`.

Reply:
Updated!

Reply:
Unfortunately, this change leads to pi_heating_demand reporting registration failing for TRV:

```
'pi_heating_demand': {'min': 30, 'max': 900, 'id': 'pi_heating_dem
and', 'name': 'pi_heating_demand', 'change': 5, 'status': 'INVALID_DATA_TYPE'}}
``` 

I've reverted it in order to get back change updates from device.